vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -45,6 +47,20 @@ def hosts_ports
         @hosts_ports ||= @baremetal_service.ports.details
       end
 
+      def resources
+        return @resources if @resources
+
+        resources = []
+        @orchestration_service.stacks.each do |stack|
+          # Nested depth 50 just for sure, although nobody should nest templates that much
+          all_stack_resources = @orchestration_service.list_resources(stack, :nested_depth => 50)
+          # Filtering just OS::Nova::Server, which is important to us for getting Purpose of the node
+          # (compute, controller, etc.).
```

Comment:
if the full heat refresh, we will collect all resources. In this context, we are interested only in those that tells us about node type